### PR TITLE
test(SyncPresenter): Adds unit tests to SyncPresenter. creates WorkManagerController

### PR DESCRIPTION
**Pull request by ferdyrod**
**JIRA ISSUE**: [ANDROAPP-2524](https://jira.dhis2.org/browse/ANDROAPP-2524)

### CODE REVIEW

**Comment by ferdyrod** [comment]
The WorkManagerController is very simple and it needs more methods and maybe a better way to call the methods that are already included.

File: app/src/main/java/org/dhis2/usescases/sync/SyncActivity.kt [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            WorkInfo.State.SUCCEEDED -> {
+                binding.metadataText.text = getString(R.string.configuration_ready)
+                Bindings.setDrawableEnd(
+                    binding.metadataText,
+                    AppCompatResources.getDrawable(this, R.drawable.animator_done)
+                )
+                presenter.getTheme()
+            }
+            else -> {
+            }
```
Comment (Balcan) [inline app/src/main/java/org/dhis2/usescases/sync/SyncActivity.kt]: We should add a comment on why the else branch is empty

File: app/src/main/java/org/dhis2/usescases/sync/SyncActivity.kt [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                binding.eventsText.text = getString(R.string.data_ready)
+                Bindings.setDrawableEnd(
+                    binding.eventsText,
+                    AppCompatResources.getDrawable(this, R.drawable.animator_done)
+                )
+                presenter.syncReservedValues()
+                startMain()
+            }
+            else -> {
+            }
```
Comment (Balcan) [inline app/src/main/java/org/dhis2/usescases/sync/SyncActivity.kt]: We should add a comment on why the else branch is empty

**Comment by JaimeToca** [comment]
hey ferdy, is it pointing to the right rule engine?

**Comment by ferdyrod** [comment]
> hey ferdy, is it pointing to the right rule engine?

@JaimeToca Yes, the rule-engine master branch was updated so this adds that link to the latest commit

**Comment by JaimeToca** [comment]
Great @ferdyrod, this PR looks great to me.  I'd test it the sync in this jira issue

**Pull request merged**